Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -208,18 +267,6 @@ class MessageManager: EngineWebDelegate {
         Logger.instance.info(message: "Message loaded with route: \(route)")
 
         currentRoute = route
-        if route == currentMessage.messageId, !messageLoaded {
-            messageLoaded = true
-            if isMessageEmbed {
-                delegate?.messageShown(message: currentMessage)
-            } else {
-                if UIApplication.shared.applicationState == .active {
-                    loadModalMessage()
-                } else {
-                    Gist.shared.removeMessageManager(instanceId: currentMessage.instanceId)
-                }
-            }
-        }
```

Comment:
This is code that should only be called for modal messages. So I moved this block into the Modal manager subclass.